proto/gz/msgs/lens.proto

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package gz.msgs;
+option java_package = "com.gz.msgs";
+option java_outer_classname = "LensProtos";
+
+/// \ingroup ignition.msgs
+/// \interface Lens
+/// \brief Information about a lens element
+
+message Lens
+{
+  /// \brief Types of lens models.
+  enum Type
+  {
+    TYPE_NOT_SPECIFIED   = 0;
+    GNOMONICAL           = 1;
+    STEREOGRAPHIC        = 2;
+    EQUIDISTANT          = 3;
+    EQUISOLID_ANGLE      = 4;
+    ORTHOGRAPHIC         = 5;
+    CUSTOM               = 6;
+  }
+
+  /// \brief Lens custom function type.
+  enum FunctionType
+  {
+    FUNCTION_NOT_SPECIFIED = 0;
+    SIN = 1;
+    TAN = 2;
+    ID  = 3;
+  }
+
+  /// \brief Lens type
+  Type type                      = 1;
+
+  /// \brief Lens scale to horizontal field of view.
+  bool scale_to_hfov             = 2;
```

Comment:
Note that this defaults to true on SDF, but to false here

Reply:
That should be fine. If SDF is used to generate the message then everything will be handled correctly. If someone creates the message by hand, then it's up to them to decide whether they are following the SDF or not.